Sources/Prometheus/MetricTypes/Histogram.swift

```diff
@@ -1,4 +1,73 @@
 import NIOConcurrencyHelpers
+import Dispatch
+
+/// Buckets are used by Histograms to bucket their values.
+///
+/// See https://prometheus.io/docs/concepts/metric_types/#Histogram
+public struct Buckets: ExpressibleByArrayLiteral {
+    public typealias ArrayLiteralElement = Double
+    
+    public init(arrayLiteral elements: Double...) {
+        self.init(elements)
+    }
+    
+    fileprivate init (_ r: [Double]) {
+        if r.count < 1 {
+            self.buckets = Buckets.defaultBuckets.buckets
+            return
+        }
+        var r = r
+        if !r.contains(Double.greatestFiniteMagnitude) {
+            r.append(Double.greatestFiniteMagnitude)
+        }
+        // TODO: Decide if we want to fix these "programmer mistakes" or assert them.
+        r.sort(by: <)
+        assert(Array(Set(r)).sorted(by: <) == r, "Buckets contain duplicate values.")
```

Comment:
I see... I'd somewhat expect people putting in custom buckets not so often -- and even if, they'd probably declare a constant somewhere rather than make them up dynamically. In that sense, perhaps it is better to crash hard here, since it's a static configuration most likely.

We should document that and explain what we do.

In other words -- i'd be fine with crashing and perhaps even this could be an  precondition?

Reply:
👍 Time to crash! For now I'll keep these as `asserts`, we can always see in the future if they should become preconditions